hibernate-core/src/main/antlr/hql.g

```diff
@@ -846,6 +846,13 @@ path
 	: identifier ( DOT^ { weakKeywords(); } identifier )*
 	;
 
+treatAsPath
+	: path
+	| i:IDENT! OPEN! p:treatAsPath AS! a:path! CLOSE! ( DOT^ path )? {i.getText().equalsIgnoreCase("treat") }? {
```

Comment:
TREATs cannot be nested.  

Also, not a fan of recognizing IDENT and then applying a semantic predicate to recognize a specific IDENT.  Why did you do that as opposed to just recognizing "treat"?